src/mlpack/tests/linear_svm_test.cpp

```diff
@@ -0,0 +1,1129 @@
+{
+  const size_t points = 1000;
+  const size_t inputSize = 5;
+  const size_t numClasses = 5;
+  const double lambda = 0.5;
+  const double delta = 1.0;
+
+  // Generate five-Gaussian dataset.
+  arma::mat identity = arma::eye<arma::mat>(5, 5);
+  GaussianDistribution g1(arma::vec("1.0 9.0 1.0 2.0 2.0"), identity);
+  GaussianDistribution g2(arma::vec("4.0 3.0 4.0 2.0 2.0"), identity);
+  GaussianDistribution g3(arma::vec("3.0 2.0 7.0 0.0 5.0"), identity);
+  GaussianDistribution g4(arma::vec("4.0 1.0 1.0 2.0 7.0"), identity);
+  GaussianDistribution g5(arma::vec("1.0 0.0 1.0 8.0 3.0"), identity);
+
+  arma::mat data(inputSize, points);
+  arma::Row<size_t> labels(points);
+
+  for (size_t i = 0; i < points / 5; i++)
+  {
+    data.col(i) = g1.Random();
+    labels(i) = 0;
+  }
+  for (size_t i = points / 5; i < (2 * points) / 5; i++)
+  {
+    data.col(i) = g2.Random();
+    labels(i) = 1;
+  }
+  for (size_t i = (2 * points) / 5; i < (3 * points) / 5; i++)
+  {
+    data.col(i) = g3.Random();
+    labels(i) = 2;
+  }
+  for (size_t i = (3 * points) / 5; i < (4 * points) / 5; i++)
+  {
+    data.col(i) = g4.Random();
+    labels(i) = 3;
+  }
+  for (size_t i = (4 * points) / 5; i < points; i++)
+  {
+    data.col(i) = g5.Random();
+    labels(i) = 4;
+  }
+
+  // Train linear svm object using L-BFGS optimizer.
+  LinearSVM<arma::mat> lsvm(data, labels, numClasses, lambda,
+      delta, ens::L_BFGS());
+
+  // Compare training accuracy to 1.
+  const double acc = lsvm.ComputeAccuracy(data, labels);
+  BOOST_REQUIRE_CLOSE(acc, 1.0, 2.0);
+
+  // Create test dataset.
+  for (size_t i = 0; i < points / 5; i++)
+  {
+    data.col(i) = g1.Random();
+    labels(i) = 0;
+  }
+  for (size_t i = points / 5; i < (2 * points) / 5; i++)
+  {
+    data.col(i) = g2.Random();
+    labels(i) = 1;
+  }
+  for (size_t i = (2 * points) / 5; i < (3 * points) / 5; i++)
+  {
+    data.col(i) = g3.Random();
+    labels(i) = 2;
+  }
+  for (size_t i = (3 * points) / 5; i < (4 * points) / 5; i++)
+  {
+    data.col(i) = g4.Random();
+    labels(i) = 3;
+  }
+  for (size_t i = (4 * points) / 5; i < points; i++)
+  {
+    data.col(i) = g5.Random();
+    labels(i) = 4;
+  }
+
+  // Compare test accuracy to 1.
+  const double testAcc = lsvm.ComputeAccuracy(data, labels);
+  BOOST_REQUIRE_CLOSE(testAcc, 1.0, 2.0);
+}
+
+/**
+ * Testing Train() in LinearSVM.
+ */
+BOOST_AUTO_TEST_CASE(LinearSVMTrainTest)
+{
+  // Test the stability of the LinearSVM.
+  arma::mat dataset = arma::randu<arma::mat>(5, 1000);
+  arma::Row<size_t> labels(1000);
+  for (size_t i = 0; i < 500; ++i)
+    labels[i] = size_t(0.0);
+  for (size_t i = 500; i < 1000; ++i)
+    labels[i] = size_t(1.0);
+
+  LinearSVM<arma::mat> lsvm(dataset.n_rows, 2);
+  LinearSVM<arma::mat> lsvm2(dataset.n_rows, 2);
+  lsvm.Parameters() = lsvm2.Parameters();
+  ens::L_BFGS optimizer;
+  lsvm.Train(dataset, labels, 2, std::move(optimizer));
+  lsvm2.Train(dataset, labels, 2, std::move(optimizer));
```

Comment:
I'm not sure what this test is testing, can you clarify please?

Reply:
It tests the stability of LinearSVM. It checks whether the parameters of both `lsvm` and `lsvm2` are close enough or not for the same `dataset` and `labels`.

Reply:
I guess I'm not sure why they would not be the same; since you're starting from the same initialization (I think), I wouldn't expect any part of the result to be different, since there's no randomness in the way you have it set up here.  So do you think the test is necessary?

Reply:
Well, since it was the first time I'm making all the tests for anything, I took reference from many existing tests, and this was inspired from one of the `SoftmaxRegression` test. They do start from same initialization, so I think this one is redundant.